K2Engineering/K2Engineering/CastBarOutput.cs

```diff
@@ -45,17 +45,19 @@ protected override void RegisterOutputParams(GH_Component.GH_OutputParamManager
         protected override void SolveInstance(IGH_DataAccess DA)
         {
             //Input
-            Object[] output = new Object[4];
+
+            //Object[] output = new Object[4];
+            DataTypes.BarData output = new DataTypes.BarData();
             DA.GetData(0, ref output);
 
             //Casting
-            int pIndexStart = (int)output[0];
-            int pIndexEnd = (int)output[1];
+            int pIndexStart = output.Index1;
```

Comment:
Your change is more readable than with indexing; was that the motivation? Seems cleaner to me.